### ISSUE TITLE
Add pool DB operation `removePools`.

### ISSUE THREAD

**Issue by jonathanknowles** [cardano-foundation/cardano-wallet#2017]
# Context

As part of the work to perform garbage collection of retired pools from the database, we need a convenient and safe way to _remove_ pools that have retired.

# Decision

Create a pool database operation `removePools`:

```hs
removePools
    :: [PoolId]
    -> stm ()
    -- ^ Remove all data relating to the specified pools.
````

# Acceptance Criteria

- [x] The pool DB _must_ provide a `removePools` operation.
- [x] The `removePools` operation _must_ remove each pool _atomically_.
- [x] The `removePools` operation _must_ write an entry to the log before removing a pool.

# Development 

# QA 

- [x] Create a property test that compares the SQLite and `MVar` implementations of `removePools` and confirms that they have the same effect, given the same inputs and initial conditions:
    - [x] After a `removePools` operation, `listPools` should not include any of the pools that were removed.

**Comment by piotr-iohk** [comment]
I guess that's this one that needs to be merged first: https://github.com/input-output-hk/cardano-wallet/pull/2038.

**Comment by jonathanknowles** [comment]
> I guess that's this one that needs to be merged first: #2038.

Yes, that's the one!

**Comment by piotr-iohk** [comment]
lgtm.
